llvm/test/CodeGen/AMDGPU/sgpr-spills-empty-block.mir

```diff
@@ -0,0 +1,1445 @@
+# NOTE: Assertions have been autogenerated by utils/update_mir_test_checks.py UTC_ARGS: --version 4
+# RUN: llc -mtriple=amdgcn-amd-amdhsa -mcpu=gfx942 -verify-machineinstrs -run-pass=si-lower-sgpr-spills -o - %s | FileCheck %s
+
+# Test that the si-lower-sgpr-spills pass does not defeference a
+# machine instruction iterator that is equal to end().
+
+---
+name:            empty_spill_block
```

Comment:
This test function is huge. Can you try to use llvm-reduce to cut this down? I can't imagine needing more than a few blocks

Reply:
Yeah, I've been trying to create a smaller version. This one was generated with llvm-reduce. I don't really like the test size either.

Reply:
Oh right, I never pushed the block reduction for MIR